### PR TITLE
ADAPT-5659 | @rebeccahongsf | add a check for unpublished trip to prevent breaking build

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- add a check for unpublished trip to prevent breaking build

# Review By (Date)
- asap

# Review Tasks

## Setup tasks and/or behavior to test

1. Navigate to 
- https://deploy-preview-678--stanford-alumni.netlify.app/travel-study/destinations/poland-2024
- https://deploy-preview-678--stanford-alumni.netlify.app/travel-study/destinations/poland-2024/notify
- https://deploy-preview-678--stanford-alumni.netlify.app/travel-study/destinations/china-2024
- https://deploy-preview-678--stanford-alumni.netlify.app/travel-study/destinations/china-2024/reserve
2. Verify that these pages do not exist (the trip pages are currently unpublished)
3. Review code

# Associated Issues and/or People
- ADAPT-5659

### CODE REVIEW

**Comment by rebeccahongsf** [comment]
Hmm. Looks like the notify form page still generates. Please hold on reviewing while I put together another check 😄

**Comment by rebeccahongsf** [comment]
@sherakama the registration form build conditional has been updated to check for the actual trip content. The notify form pages (e.g. [poland notify form](https://deploy-preview-678--stanford-alumni.netlify.app/travel-study/destinations/poland-2024/notify)) still builds but displays the white page of death due to missing information from the unpublished trip. 

I imagine that a user shouldn't be able to access the notify me page without going through the trip page but thought I'd check in — did we want to account for this use case? 

I started testing a few things (e.g. adding build conditionals for the `formPage` content type) but decided to pause since the right conditional statement taking me longer than expected to figure out 🤔

File: gatsby-node.js [hunk 1]
```diff
@@ -136,10 +136,11 @@ exports.createPages = ({ graphql, actions }) => {
           ) {
             isCanonical = false;
           }
+
           const noIndex = content.noIndex ? content.noIndex : false;
 
           // Create the GG form page(s) only if the trip is published
-          if (content.trip) {
+          if (content.trip && content.trip.content?.title) {
```
Comment (sherakama) [inline gatsby-node.js:143]: So, when a trip is selected but unpublished, `content.trip` is the string id of the story in storyblok. When the trip is selected and published, `content.trip` is an object with stuff like `content.trip.content`. When no trip is selected, content.trip is undefined. 

So, you could check that `content.trip` exists, is an object, and has content.

**Review by sherakama — CHANGES_REQUESTED** [review]
Please revise the validation.

**Comment by sherakama** [comment]
> I imagine that a user shouldn't be able to access the notify me page without going through the trip page but thought I'd check in — did we want to account for this use case?

It breaks because of missing information in the destructure statement on line 24. That should be a quick fix to handle and produce an error notice vs white screen. Please handle this case as well.

**Comment by sherakama** [comment]
FYI: I published the China trip to get another build to work. If you start working on this again you will need to unpublish https://app.storyblok.com/#/me/spaces/118608/stories/0/0/272989885

**Review by sherakama — APPROVED** [review]
Thanks for sticking with this @rebeccahongsf 

This is GTG for me now.

**Pull request merged**